packages/sw-cache-expiration/test/sw/plugin.js

```diff
@@ -0,0 +1,226 @@
+);
+
+const expect = self.chai.expect;
+mocha.setup({
+  ui: 'bdd',
+  reporter: null,
+});
+
+let count = 0;
+function getUniqueUrl() {
+  return 'https://example.com/?count=' + count++;
+}
+function getUniqueCacheName() {
+  return 'test-cache-' + count++;
+}
+
+describe('Test of the Plugin class', function() {
+  const maxAgeSeconds = 3;
+  const maxEntries = 3;
+  const now = 1487106334920;
+  const timestampPropertyName = goog.cacheExpiration.timestampPropertyName;
+  const urlPropertyName = goog.cacheExpiration.urlPropertyName;
+
+  const Plugin = goog.cacheExpiration.Plugin;
+
+  it(`should throw when Plugin() is called without any parameters`, function() {
+    let thrownError = null;
+    try {
+      new Plugin();
+    } catch(err) {
+      thrownError = err;
+    }
+
+    expect(thrownError).to.exist;
+    expect(thrownError.name).to.equal('max-entries-or-age-required');
+  });
+
+  it(`should throw when Plugin() is called with an invalid maxEntries parameter`, function() {
+    let thrownError = null;
+    try {
+      new Plugin({maxEntries: 'invalid'});
+    } catch(err) {
+      thrownError = err;
+    }
+    expect(thrownError).to.exist;
+    expect(thrownError.name).to.equal('max-entries-must-be-number');
+  });
+
+  it(`should throw when Plugin() is called with an invalid maxAgeSeconds parameter`, function() {
+    let thrownError = null;
+    try {
+      new Plugin({maxAgeSeconds: 'invalid'});
+    } catch(err) {
+      thrownError = err;
+    }
+    expect(thrownError).to.exist;
+    expect(thrownError.name).to.equal('max-age-seconds-must-be-number');
+  });
+
+  it(`should use the maxAgeSeconds from the constructor`, function() {
+    const plugin = new Plugin({maxAgeSeconds});
+    expect(plugin.maxAgeSeconds).to.equal(maxAgeSeconds);
+  });
+
+  it(`should use the maxEntries from the constructor`, function() {
+    const plugin = new Plugin({maxEntries});
+    expect(plugin.maxEntries).to.equal(maxEntries);
+  });
+
+  it(`should return the same IDB instance when getDB() is called multiple times`, function() {
+    const cacheName = getUniqueCacheName();
+    const plugin = new Plugin({maxEntries});
+    return plugin.getDB({cacheName}).then((firstDB) => {
+      return plugin.getDB({cacheName}).then((secondDB) => {
+        expect(firstDB).to.eql(secondDB);
+      });
+    });
+  });
+
+  it(`should return the same Cache instance when getCache() is called multiple times`, function() {
+    const cacheName = getUniqueCacheName();
+    const plugin = new Plugin({maxEntries});
+    return plugin.getCache({cacheName}).then((firstCache) => {
+      return plugin.getCache({cacheName}).then((secondCache) => {
+        expect(firstCache).to.eql(secondCache);
+      });
+    });
+  });
+
+  it(`should return true when there's no cachedResponse passed to isResponseFresh()`, function() {
+    const plugin = new Plugin({maxAgeSeconds});
+    expect(plugin.isResponseFresh()).to.be.true;
```

Comment:
The underlying method gets called as part of the `RequestWrapper` lifecycle callback `cacheWillMatch`. `cacheWillMatch` might be called with an undefined `Response` passed in if there isn't yet an entry in the cache for a given URL, and we just want to return `true` in that case to work with the semantics that `RequestWrapper` expects.